src/components/ExperienceForm/ExperienceForm.js

```diff
@@ -3,155 +3,222 @@
+  handleChange: PropTypes.func.isRequired,
+  handleBlur: PropTypes.func.isRequired,
+  isSubmitting: PropTypes.bool,
+  values: PropTypes.shape({
+    title: PropTypes.string,
+    body: PropTypes.string,
+    field_experience_path: PropTypes.string
+  }).isRequired,
+  errors: PropTypes.shape({
+    title: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
+    field_experience_path: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.bool
+    ]),
+    body: PropTypes.oneOfType([PropTypes.string, PropTypes.bool])
+  }).isRequired,
+  touched: PropTypes.shape({
+    title: PropTypes.bool,
+    field_experience_path: PropTypes.bool,
+    body: PropTypes.bool
+  }).isRequired,
+  classes: PropTypes.shape({
+    textField: PropTypes.string.isRequired,
+    button: PropTypes.string.isRequired
+  }).isRequired,
+  user: PropTypes.shape({
+    authentication: PropTypes.shape({
+      accessToken: PropTypes.string.isRequired,
+      csrfToken: PropTypes.string.isRequired
+    }).isRequired
+  }).isRequired,
+  experiences: PropTypes.shape({
+    error: PropTypes.string,
+    items: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.string.isRequired,
+        title: PropTypes.string.isRequired,
+        body: PropTypes.PropTypes.shape({
+          value: PropTypes.string
+        }),
+        field_experience_path: PropTypes.string.isRequired
+      })
+    )
+  })
+};
 
-  /**
-   * Fetches the experience that belongs to the given slug.
-   */
-  getExperience = () => {
-    const {
-      experienceSlug,
-      experiences: { items }
-    } = this.props;
-    return experienceSlug
-      ? items.find(item => item.field_experience_path === experienceSlug)
-      : false;
-  };
+ExperienceForm.defaultProps = {
+  isSubmitting: false,
+  experiences: {
+    error: null,
+    items: []
+  }
+};
 
-  /**
-   * Handles dispatching create or update actions.
-   *
-   * @param {object} event - Submit event object.
-   */
-  handleSubmit = event => {
-    event.preventDefault();
+const FormikExperienceForm = withFormik({
+  displayName: 'ExperienceForm',
+  enableReinitialize: true,
+  mapPropsToValues: props => {
+    const { experienceSlug, experiences } = props;
+    const experience = parseExperience(experiences, experienceSlug);
+
+    const values = {
+      title: '',
+      field_experience_path: '',
+      body: ''
+    };
+
+    if (experience) {
+      const { title, field_experience_path, body } = experience;
+      Object.assign(values, {
+        title,
+        field_experience_path,
+        body: body ? body.value : ''
+      });
+    }
+
+    return values;
+  },
+  validationSchema: object().shape({
+    title: string()
+      .required()
+      .min(3)
+      .max(50),
+    body: string()
+      .required()
+      .min(3)
+      .max(200),
+    field_experience_path: string()
+      .required()
+      .min(3)
+      .max(100)
+      .matches(FORM_INPUT_VALIDATION_REGEX_URL_SEGMENT, {
```

Comment:
Made this regex into a const since it's already used in multiple places, and likely to be used in a couple other places.